src/storage/upgrade_project.ts

```diff
@@ -121,17 +121,10 @@ async function upgradeFrom_002_to_003(
       try {
         Blockly.serialization.workspaces.load(blocks, headlessWorkspace);
 
-        // Find and modify init method blocks to remove robot parameter
-        const allBlocks = headlessWorkspace.getAllBlocks();
-        for (const block of allBlocks) {
-          if (block.type === MRC_CLASS_METHOD_DEF_BLOCK_NAME &&
-            block.getFieldValue('NAME') === 'init') {
-            // Remove robot parameter from init method
-            const methodBlock = block as ClassMethodDefBlock;
-            let filteredParams: Parameter[] = methodBlock.mrcParameters.filter(param => param.name !== 'robot');
-            methodBlock.mrcParameters = filteredParams;
-          }
-        }
+        // Method blocks need to be upgraded
+        headlessWorkspace.getBlocksByType(MRC_CLASS_METHOD_DEF_BLOCK_NAME, false).forEach(block => {
```

Comment:
Can you move lines 125-127 to a new function (outside of the block class) in mrc_class_method_def.ts?

Reply:
I think it is worse because it makes it more difficult to abstract the upgrade in the future.   I can easily imagine a method that takes a list of block types that need to be upgraded for a certain upgrade and it iterates through all of them calling the method for upgrading.

But I may be missing something and want to understand.

Reply:
OK. That's a good point.